packages/vite/src/node/config.ts

```diff
@@ -1859,18 +1859,20 @@ async function loadConfigFromBundledFile(
   // write it to disk, load it with native Node ESM, then delete the file.
   if (isESM) {
```

Comment:
```ts
const hasNpmPrefixExternalImport = Object.values(result.metafile.outputs)
    .flatMap((output) => output.imports)
    .filter((imp) => imp.external && imp.path.startsWith('npm:'))
```
Does passing this value from `bundleConfigFile` and only avoid writing to `node_modules` in that case work? If I understand correctly, the problem only happens if the config file has an import with `npm:` prefix.

If you have a code like below, it won't work. But this is already not supported, so it should be fine.
```ts
const dynImport = (str: string) => import(str)

dynImport('npm:foo')
```

Reply:
> If I understand correctly, the problem only happens if the config file has an import with npm: prefix.

Yes, you're right! Here's a break down of what's going to happen for clarity.

1. Vite removes the `npm:` prefix from import statements 
2. bundled file is executed with Deno
3. Deno's module resolution expects an `npm:` prefix when we want to do something like `import { defineConfig } from 'vite';` 
4. However since this bundled file is placed in `node_modules/`, Deno will consider the file as an Node.js compatible module and resolve import statements accordingly 
5. import works!

So your solution should work, but I think it make thing too unpredictable and fragile because it's like trying to implement a transformation layer that voids Deno's behavior difference for modules inside/outside of `node_modules/`

Since the bundled file is going to be executed by Deno not Node, I'd say that keeping the import statements as-is  is a better solution.